### PR TITLE
Calculator user Interface

### PR DESCRIPTION
**Pull request by vidhishweta01**
added - 
button.js
outputScreen.js
calculator.js
styled all the component

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import './App.css';
+import React from 'react';
+import Calculator from './calculator';
```
Comment (bellom) [inline src/App.js]: Let's create the Display component and import it here...

image

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
 
 function App() {
   return (
     <div className="App">
```
Comment (bellom) [inline src/App.js]: **Please let's stick to the project requirement, kindy double-check if you're following the instructions in the pathwright, and also do well to follow the naming conversion.**

We need to render the Display and ButtonPanel components in `App.js`, and also you need to wrap both `Display` and `ButtonPanel` with [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax)
Screenshot from pathwright:
image


Reason for making use of React Fragment:
We don't want to create an extra element, that's why you need to use fragment, so we are able to render without having to create an extra node wrapping `Display` and `ButtonPanel`.

File: src/button.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import styles from './component.module.css';
+
+const Button = ({ label }) => <button type="button" className={styles.buton}>{label}</button>;
```
Comment (bellom) [inline src/button.js]: For naming conversion, it would be nice to use `buttonName` instead of `label`

File: src/calculator.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import CalculatorTitle from './calculatorTitle';
+import OutputScreen from './outputScreen';
+import Button from './button';
+
+const Calculator = () => (
```
Comment (bellom) [inline src/calculator.js]: We should update the component name to `ButtonPanel`

File: src/calculator.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  <div className="frame">
+    <CalculatorTitle value="Calculator" />
+    <div className="mainCalc">
+      <OutputScreen />
+      <div className="button-row">
+        <Button label="Clear" />
+        <Button label="Delete" />
```
Comment (bellom) [inline src/calculator.js]: Kindly create calculator panel with all the buttons in the following groups: 👇 

image

🤔 `Clear` and `delete` should not be inside the group.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @vidhishweta01  :clap: 

### Awesome job so far!! :+1: 

# Status: Changes Required! ♻️ 

**Kindly check my comments under this review;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

File: src/component/App/App.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import styles from './App.module.css';
+import ButtonPanel from '../buttonPanel';
+
+function App() {
+  return (
+    <div className={styles.App}>
+      <ButtonPanel />
+    </div>
```
Comment (rindrajosia) [inline src/component/App/App.js]: - [x] [REQUIRED] it should also render the `Display` components  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/appFunctionalComponent.png))
- [x] [REQUIRED] `Display` and `ButtonPanel` components should be wrapped in [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax) short syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/)) not in a `DIV`

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/component/App/App.js [hunk 7]
```diff
@@ -1,12 +1,14 @@
...
+import Display from '../Display';
 
 function App() {
   return (
-    <div className={styles.App}>
+    <React.Fragment className="App">
+      <Display />
       <ButtonPanel />
-    </div>
+    </React.Fragment>
```
Comment (whiz25) [inline src/component/App/App.js]: Please use the shorthand version: `<></>`

**Review by whiz25 — CHANGES_REQUESTED** [review]
## changes required 🙏🏿 🙏🏿 🙏🏿 
Nice work making the requested changes. We need to use react fragment shorthand syntax.
Check my comment.


### Martin Afani
#### Slack: whiz25

File: src/component/App/App.js [hunk 8]
```diff
@@ -1,14 +1,16 @@
...
-    <React.Fragment className="App">
-      <Display />
-      <ButtonPanel />
-    </React.Fragment>
+    <>
+      <div className={styles.App}>
+        <Display />
+        <ButtonPanel />
+      </div>
+    </>
```
Comment (rindrajosia) [inline src/component/App/App.js:13]: - [ ] [REQUIRED] What you have done here is similar to:
```JS
<div className={styles.App}>
        <Display />
        <ButtonPanel />
 </div>
```
What you have to do is:
```JS
<>
   <Display />
   <ButtonPanel />
</>
```

Don't pay attention to the style in this milestone. You will work on styling in `milestone 3`

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------

File: src/component/App/App.js (hunk 8 above)
Comment (vidhishweta01) [reply src/component/App/App.js:13]: so you want me to remove it now is it written somewhere that If something is styled you have to remove it

Comment (vidhishweta01) [reply src/component/App/App.js:13]: Its only aligning the items

Comment (vidhishweta01) [reply src/component/App/App.js:13]: If it is styled is it making any logical errors

File: src/component/buttonPanel/buttonPanel.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    </div>
+    <div className={styles.button_row}>
+      <Button label="0" />
+      <Button label="." />
+      <Button label="=" />
+    </div>
+  </div>
+);
+
+// Export Calculator Component.
```
Comment (Elbie-em) [inline src/component/buttonPanel/buttonPanel.js:39]: #### [OPTIONAL]

- [ ] Kindly remove this comment, the export below clearly states you are exporting the component.

File: src/component/buttonPanel/buttonPanel.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import Button from '../button';
+import styles from './buttonPanel.module.css';
+
+const Calculator = () => (
```
Comment (Elbie-em) [inline src/component/buttonPanel/buttonPanel.js:5]: #### [OPTIONAL]

- [ ] For naming convention purposes I highly recommend you change this component's name to `ButtonPanel` as required in the project requirements.

File: src/component/App/App.module.css [hunk 11]
```diff
@@ -0,0 +1,7 @@
+.App {
+  text-align: center;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  width: 100%;
+}
```
Comment (Elbie-em) [inline src/component/App/App.module.css:7]: #### [OPTIONAL]

- [ ] I highly recommend that you move all styling for your components to an isolated stylesheet folder, the component folder should only consist of components and you can simply import your styles to their respective components.

**Review by Elbie-em — APPROVED** [review]
## STATUS: APPROVED ✅ 

Hi @vidhishweta01, great job on applying the changes requested by a previous reviewer 👏  , I am therefore approving this milestone, keep on with the hard work.

## Optional 

**_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

**Pull request merged**